### PR TITLE
Parallelized IHDG

### PR DESCRIPTION
**Pull request by kazbekkm**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,13 @@
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -g")
 
-add_executable(
+#[[add_executable(
```
Comment (bremerm31) [inline test/CMakeLists.txt]: Can we uncomment these tests?

File: source/general_definitions.hpp [hunk 2]
```diff
@@ -103,10 +103,8 @@ class Basis {
      */
     virtual DynMatrix<double> GetMinv(const uint p) = 0;
 
-    template <typename InputArrayType>
-    decltype(auto) ProjectBasisToLinear(const InputArrayType& u);
-    template <typename InputArrayType>
-    decltype(auto) ProjectLinearToBasis(const uint ndof, const InputArrayType& u_lin);
+    virtual DynMatrix<double> ProjectBasisToLinear(const DynMatrix<double>& u)                      = 0;
```
Comment (bremerm31) [inline source/general_definitions.hpp]: Why do we need this?

I'd prefer to keep the expression templates intact as these projections get used in slope limiting and wetting and drying.

File: test/CMakeLists.txt (hunk 1 above)
Comment (kazbekkm) [reply test/CMakeLists.txt]: I have changed some stuff in data structure, so these test will not compile right. :) I think I'll fix them or remove them altogether.

Comment (bremerm31) [reply test/CMakeLists.txt]: :+1: for fixing

I can help if need be. Also might make sense to split them into a simple Eigen, Blaze, and PETSC tests.

File: source/general_definitions.hpp (hunk 2 above)
Comment (kazbekkm) [reply source/general_definitions.hpp]: If you look at previous implementation, I had to create a new DynMatrix with projected values and return. So there are no expression templates going on in these function. I though about a way to use expression templates and stuff. I need to store a matrix in basis class and I though maybe that's not the best option.

**Comment by kazbekkm** [comment]
[TEst.xlsx](https://github.com/UT-CHG/dgswemv2/files/2470854/TEst.xlsx)

So fork join is bad and has been removed

**Pull request merged**